src/IHttpEndpoint.ts

```diff
@@ -1,4 +1,8 @@
+import { executionAsyncResource } from 'async_hooks';
 import { APIGatewayProxyEventV2 } from 'aws-lambda';
 import { IHandler } from './IHandler';
 
-export interface IHttpEndpoint<Request> extends IHandler<Request, APIGatewayProxyEventV2> {}
\ No newline at end of file
+export interface IHttpEndpoint<ExecuteQueryRequest, ExecuteQueryResults> extends IHandler<Request, APIGatewayProxyEventV2> 

```

Comment:
Try to be VERY general/abstract with names for generic params. I chose Input/Output because they would help indicate the usage (input model, output model). 

The type names you used here make it look those are actual types that are defined somewhere else (they're too specific).